tools/importer-rest-api-specs/components/parser/flattener.go

```diff
@@ -85,38 +85,108 @@ func load(directory string, fileName string) (*SwaggerDefinition, error) {
 	}, nil
 }
 
-func findAndMergeLocalMixins(input *loads.Document, basePath string, baseFile string) (*loads.Document, error) {
-	if len(strings.Split(baseFile, "/")) != 2 { // We only care about local files, not sub-folders
-		return input, nil
+func findLocalMixins(input *loads.Document, filePath string) (*spec.Swagger, error) {
+	mixins := &spec.Swagger{
+		SwaggerProps: spec.SwaggerProps{
+			Definitions: make(map[string]spec.Schema),
+		},
 	}
-	pathsToMixin := make([]string, 0)
+
 	if input.Analyzer != nil {
-		allRefs := input.Analyzer.AllRefs()
-		for _, v := range allRefs {
-			if path := v.Ref.GetURL(); path != nil && path.Path != "" && len(strings.Split(path.Path, "/")) == 2 { // Check if we have a reference in the CWD.
-				pathsToMixin = append(pathsToMixin, path.Path)
+		for _, ref := range input.Analyzer.AllRefs() {
+			if err := findMixinForRef(ref, filePath, input, mixins); err != nil {
+				return nil, fmt.Errorf("find mixin for ref %q in %q: %+v", ref.String(), filePath, err)
 			}
 		}
 	}
 
-	if len(pathsToMixin) > 0 {
-		uniqueFilter := make(map[string]bool)
-		uniquePaths := make([]string, 0)
-		for _, path := range pathsToMixin {
-			if _, ok := uniqueFilter[path]; !ok {
-				uniqueFilter[path] = true
-				uniquePaths = append(uniquePaths, path)
+	return mixins, nil
+}
+
+func findMixinForRef(ref spec.Ref, filePath string, doc *loads.Document, mixins *spec.Swagger) error {
+	if path := ref.GetURL(); path == nil || (path.Path != "" && len(strings.Split(path.Path, "/")) != 2) {
+		// Check if we have a reference in the CWD. if path.Path is empty string, it refers to the same file
+		return nil
+	}
```

Comment:
this is to be consistent with current logic which only resolves ref in CWD, but for datafactory, we should expand the scope to sub directory.